src/main/java/UseCase/NextSong.java

```diff
@@ -0,0 +1,39 @@
+package UseCase;
+
+import Entities.MusicPlayer;
+import Entities.Song;
+import InputBoundary.NextSongInputBoundary;
+import InputBoundary.playPlaylistInputBoundary;
+import InputData.playlistInputData;
+import OutputBoundary.songOutputBoundary;
+
+import java.util.ArrayList;
+
+public class NextSong implements NextSongInputBoundary {
+    private final ArrayList<Song> playlist;
+    private final String name;
+    private final songOutputBoundary presenter;
+
+    public NextSong(playlistInputData data, songOutputBoundary presenter){
+        this.playlist = data.getSongs();
+        this.presenter = presenter;
+        this.name = data.getName();
+    }
+
+    @Override
+    public void skipSong() {
+        MusicPlayer mp = MusicPlayer.getInstance();
+        int id;
+        if (mp.getCurrentSong() != null)
+            id = playlist.indexOf(mp.getCurrentSong());
+        else {
+            mp.close();
+            presenter.songPlayed(null);
```

Comment:
Could you add some comments explaining your logic here? I'm not sure I follow what/why the else branch is doing.

Reply:
I commented the index of line, but I hesitate to add one to the else branch to avoid a code smell. basically, the else branch (which I've now slightly modified to check another condition - if the index is the last one in the playlist) only occurs if skipSong() is somehow called when there is no playlist playing, so it just closes the media player, and returns null to the presenter which I will handle later when I implement the play bar.